addons/l10n_it_stock_ddt/tests/test_ddt.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo.addons.sale.tests.test_sale_common import TestSale
+from odoo.tests import tagged
+
+
+@tagged('post_install', '-at_install')
+class TestDDT(TestSale):
+
+    def test_00_ddt_flow(self):
+        """
+            We confirm a sale order and handle its delivery partially.
+            This should have created a DDT number and when we generate and the invoice,
+            the delivery should be linked to it as DDT.
+        """
+        self.so = self.env['sale.order'].create({
+            'partner_id': self.partner.id,
+            'partner_invoice_id': self.partner.id,
+            'partner_shipping_id': self.partner.id,
+            'order_line': [(0, 0, {'name': p.name, 'product_id': p.id, 'product_uom_qty': 2, 'product_uom': p.uom_id.id,
+                                   'price_unit': p.list_price, 'tax_id': self.env.company.account_sale_tax_id}) for p in self.products.values()],
+            'pricelist_id': self.env.ref('product.list0').id,
+            'picking_policy': 'direct',
+        })
+        self.so.action_confirm()
+
+        # deliver partially
+        pick = self.so.picking_ids
+        pick.move_lines.write({'quantity_done': 1})
+        wiz_act = pick.button_validate()
+        wiz = self.env[wiz_act['res_model']].browse(wiz_act['res_id'])
+        wiz.process()
+
+        self.assertTrue(pick.l10n_it_ddt_number, 'The outgoing picking should have a DDT number')
+        self.inv1 = self.so._create_invoices()
+        self.inv1.action_post()
+        self.assertEqual(self.inv1.l10n_it_ddt_ids.ids, pick.ids, 'DDT should be linked to the invoice')
```

Comment:
the use case is a bit simple, I think it should test also with several lines, different pickings, several invoices etc.